R/data.table.R

```diff
@@ -1739,22 +1739,48 @@ replace_dot_alias = function(e) {
         GForce = FALSE
       } else {
         # Apply GForce
+        noCall_noVars = function(expr) (!is.call(expr) || length(all.vars(expr, max.names=1L))==0) && !dotN(expr)
+        .gshift_ok = function(q) {
+          q_named = match.call(shift, q)
+          noCall_noVars(q_named[["n"]]) &&
+            noCall_noVars(q_named[["fill"]]) &&
+            noCall_noVars(q_named[["type"]]) &&
+            is.null(q_named[["give.names"]])
+        }
+        .ghead_ok = function(q) {
+          length(q)==3L &&
+            length(q[[3L]])==1L &&
```

Comment:
should the check `length(e) == 1` be subsumed into `noCall_noVars`? maybe as an argument?